### DISCUSSION TITLE
OpenIDConnect provider's HTTPS certificate doesn't match configured thumbprint

### DISCUSSION THREAD

**Discussion by erik-dreamteam** [community/community#59455] [Actions] [answered]
### Select Topic Area

Bug

### Body

This has now cropped up twice in the last week, where our action has just started failing arbitrarily for an hour or longer.  This may not be the right place for the issue, if it's not, hopefully someone will be kind enough to help redirect.

```
with:
    role-to-assume: ***
    aws-region: us-west-2
    audience: sts.amazonaws.com
 NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)
Error: OpenIDConnect provider's HTTPS certificate doesn't match configured thumbprint
```

Based on all the docs, we should be adding the intermediate certificate to the AWS Identity Provider to verify the token.

```
HOST=$(curl https://vstoken.actions.githubusercontent.com/.well-known/openid-configuration | jq -r '.jwks_uri | split("/")[2]')
echo | openssl s_client -servername $HOST -showcerts -connect $HOST:443 2> /dev/null \
| sed -n -e '/BEGIN/h' -e '/BEGIN/,/END/H' -e '$x' -e '$p' | tail +2 \
| openssl x509 -fingerprint -sha1 -noout \
| sed -e "s/.*=//" -e "s/://g" \
| tr "ABCDEF" "abcdef"
```
Results in `6938fd4d98bab03faadb97b34396831e3780aea1`, which is the same value that's been there for over 6 months.  I figured I'd start here as there was an issue back in November that had the vstoken certificates get rotated/new intermediate was used.

**Comment by mchineboy** [comment]
Seeing the same issue here as well, and the thumbprint matches what we have stored.

**Comment by bendrucker** [comment, marked as answer]
See https://github.blog/changelog/2023-06-27-github-actions-update-on-oidc-integration-with-aws/ for the necessary remediation. 

If you run `openssl s_client` in a loop you'll see that there are two different CA certificates returned at random:

```sh
openssl s_client -servername token.actions.githubusercontent.com -showcerts -connect token.actions.githubusercontent.com:443
```

The AWS documentation assumes that the server always uses the same CA certificate, which is no longer the case:

https://github.com/aws-actions/configure-aws-credentials/issues/357#issuecomment-1610207576

The real answer here would be for GitHub to expose these fingerprints through the [meta API](https://docs.github.com/en/rest/meta/meta?apiVersion=2022-11-28).

**Reply by erik-dreamteam** [reply]
Glad they wrote something about it, wasn't there when I posted the issue this morning.  Agreed that there's a better solution out there that has them provide a data source of what the valid intermediates are.
